tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs

```diff
@@ -91,48 +98,56 @@ public static class SelfInitializedFakesSpecs
         {
-            "establish"
+            "Given a path that does not exist"
+                .x(() => fileRecorderPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()));
+
+            "And a real service to wrap while recording"
                 .x(() =>
-                    {
-                        fileRecorderPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                {
+                    realServiceWhileRecording = A.Fake<ILibraryService>();
 
-                        realServiceWhileRecording = A.Fake<ILibraryService>();
-                        realServiceDuringPlayback = A.Fake<ILibraryService>();
+                    A.CallTo(() => realServiceWhileRecording.GetCount("8"))
+                        .Returns(0x8);
+                });
 
-                        A.CallTo(() => realServiceWhileRecording.GetCount("8")).Returns(8);
-                    });
+            "And a real service to wrap while playing back"
+                .x(() => realServiceDuringPlayback = A.Fake<ILibraryService>());
 
-            "when self initializing a fake with a file recorder"
+            "When I use a self-initialized fake recording to the path to get the count for book 8"
                 .x(() =>
+                {
+                    try
                     {
-                        try
-                        {
-                            using (var recorder = Recorders.FileRecorder(fileRecorderPath))
-                            {
-                                var fakeService = A.Fake<ILibraryService>(options => options
-                                    .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
-                                countWhileRecording = fakeService.GetCount("8");
-                            }
-
-                            using (var recorder = Recorders.FileRecorder(fileRecorderPath))
-                            {
-                                var playbackFakeService = A.Fake<ILibraryService>(options => options
-                                    .Wrapping(realServiceDuringPlayback).RecordedBy(recorder));
-
-                                countDuringPlayback = playbackFakeService.GetCount("8");
-                            }
-                        }
-                        finally
+                        using (var recorder = Recorders.FileRecorder(fileRecorderPath))
                         {
-                            File.Delete(fileRecorderPath);
+                            var fakeService = A.Fake<ILibraryService>(options => options
+                                .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
+                            countWhileRecording = fakeService.GetCount("8");
                         }
-                    })
+                    }
+                    catch
+                    {
+                        File.Delete(fileRecorderPath);
+                    }
+                })

```

Comment:
I'm back. Turns out the Teardown [should execute](https://github.com/xbehave/xbehave.net/blob/ed11fcdda746edb9a111ac1b109d4d37f13a7c2a/tests/Xbehave.Test.Acceptance.Net40/TeardownFeature.cs#L146-L161) anyhow!